lib/compat/wordpress-5.9/class-wp-rest-global-styles-controller.php

```diff
@@ -0,0 +1,588 @@
+						'type'        => array( 'object' ),
+						'context'     => array( 'view', 'edit' ),
+					),
+					'settings' => array(
+						'description' => __( 'Global settings.' ),
+						'type'        => array( 'object' ),
+						'context'     => array( 'view', 'edit' ),
+					),
+					'title'    => array(
+						'description' => __( 'Title of the global styles variation.' ),
+						'type'        => array( 'object', 'string' ),
+						'default'     => '',
+						'context'     => array( 'embed', 'view', 'edit' ),
+						'properties'  => array(
+							'raw'      => array(
+								'description' => __( 'Title for the global styles variation, as it exists in the database.' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit', 'embed' ),
+							),
+							'rendered' => array(
+								'description' => __( 'HTML title for the post, transformed for display.' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+						),
+					),
+				),
+			);
+
+			$this->schema = $schema;
+
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		/**
+		 * Checks if a given request has access to read a single theme global styles config.
+		 *
+		 * @since 5.9.0
+		 *
+		 * @param WP_REST_Request $request Full details about the request.
+		 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+		 */
+		public function get_theme_item_permissions_check( $request ) {
+			// Verify if the current user has edit_theme_options capability.
+			// This capability is required to edit/view/delete templates.
+			if ( ! current_user_can( 'edit_theme_options' ) ) {
+				return new WP_Error(
+					'rest_cannot_manage_global_styles',
+					__( 'Sorry, you are not allowed to access the global styles on this site.' ),
+					array(
+						'status' => rest_authorization_required_code(),
+					)
+				);
+			}
+
+			return true;
+		}
+
+		/**
+		 * Returns the given theme global styles config.
+		 *
+		 * @since 5.9.0
+		 *
+		 * @param WP_REST_Request $request The request instance.
+		 * @return WP_REST_Response|WP_Error
+		 */
+		public function get_theme_item( $request ) {
+			if ( wp_get_theme()->get_stylesheet() !== $request['stylesheet'] ) {
+				// This endpoint only supports the active theme for now.
+				return new WP_Error(
+					'rest_theme_not_found',
+					__( 'Theme not found.' ),
+					array( 'status' => 404 )
+				);
+			}
+
+			$theme  = WP_Theme_JSON_Resolver::get_merged_data( 'theme' );
+			$data   = array();
+			$fields = $this->get_fields_for_response( $request );
+
+			if ( rest_is_field_included( 'settings', $fields ) ) {
+				$data['settings'] = $theme->get_settings();
+			}
+
+			if ( rest_is_field_included( 'styles', $fields ) ) {
+				$raw_data       = $theme->get_raw_data();
+				$data['styles'] = isset( $raw_data['styles'] ) ? $raw_data['styles'] : array();
```

Comment:
This in core is `$theme->get_raw_data()['styles'];` assuming that every `theme.json` has `styles` property. We'll need to backport to core. What I'm wondering though should we set this to empty array or not set at all? @oandregal , @jorgefilipecosta

Reply:
Core patch here: https://github.com/WordPress/wordpress-develop/pull/2224